packages/lego-bricks/src/components/Tabs/TabContainer.tsx

```diff
@@ -23,15 +49,62 @@ export const TabContainer = ({ className, lineColor, children }: Props) => {
         width: `${width}px`,
         transform: `translateX(${left}px)`,
       });
+
+      // Scroll into view
+      activeTab.scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest',
+        inline: 'nearest',
+      });
     }
+    checkScroll();
   }, [children, location.pathname]);
 
+  useEffect(() => {
+    const element = tabListRef.current;
+    if (element) {
+      element.addEventListener('scroll', checkScroll);
+      window.addEventListener('resize', checkScroll);
+      checkScroll();
+    }
+
+    return () => {
+      element?.removeEventListener('scroll', checkScroll);
+      window.removeEventListener('resize', checkScroll);
+    };
+  }, []);
```

Comment:
```suggestion
      checkScroll();

      return () => {
        element.removeEventListener('scroll', checkScroll);
        window.removeEventListener('resize', checkScroll);
      };
    }
  }, []);
```
Move the event listener removal into the if statement to make sure to only remove listeners we actually added.

Reply:
Wouldn't it be safer to run the cleanup function regardless of whether the element exists or not during unmount? I suppose there's always a chance of `element` becoming falsy during the lifecycle